stream/src/placeholder/utils/network.rs

```diff
@@ -0,0 +1,67 @@
+use super::super::Error;
+use bytes::Bytes;
+use commonware_runtime::{Sink, Stream};
+
+// TODO: Test this and write a suitable comment.
+const CHUNK_PADDING: usize = 16;
+
+pub async fn send_next_frame(
+    sink: &mut dyn Sink,
+    buf: &[u8],
+) -> Result<(), Error> {
+    // Send the length of the message
+    let len = buf.len() as u32;
+    sink.send(&len.to_be_bytes())
+        .await
+        .map_err(|_| Error::SendFailed)?;
+
+    // Send the rest of the message
+    sink.send(buf)
+        .await
+        .map_err(|_| Error::SendFailed)?;
+
+    Ok(())
+}
+
+pub async fn read_next_frame<T: Stream>(
+    stream: &mut T,
+    max_message_size: usize,
+) -> Result<Bytes, Error> {
+    // Read the first 4 bytes to get the length of the message
+    let mut buf = [0u8; 4];
+    let n = stream.recv(&mut buf)
+        .await
+        .map_err(|_| Error::StreamClosed)?;
+
+    // Validate that the frame size was read correctly
+    if n != 4 {
+        return Err(Error::ReadMalformed);
+    }
+
+    // Validate frame size
+    let len = u32::from_be_bytes(buf) as usize;
+    if len > max_message_size + CHUNK_PADDING {
+        return Err(Error::ReadTooLarge(len));
+    }
+    if len == 0 {
+        return Err(Error::StreamClosed);
+    }
+
+    // Read the rest of the message
+    let mut buf = vec![0u8; len];
+    let n = stream.recv(&mut buf)
+        .await
+        .map_err(|_| Error::ReadFailed)?;
+
+    // Validate that the message was read correctly
+    if n != len {
+        return Err(Error::ReadMalformed);
+    }
+
+    // Read message
+    let msg = self.stream.recv()
+        .await
+        .map_err(|_| Error::StreamClosed)?;
+
+    Ok(Bytes::from(buf))
+}
```

Comment:
For testing, I already implemented mocks for `Sink/Stream` that should make testing this straightforward.

https://github.com/commonwarexyz/monorepo/blob/main/runtime/src/mocks.rs